modules/Microsoft.RecoveryServices/vaults/deploy.bicep

```diff
@@ -173,6 +176,7 @@ resource rsv 'Microsoft.RecoveryServices/vaults@2022-09-10' = {
   properties: {
     monitoringSettings: !empty(monitoringSettings) ? monitoringSettings : null
     securitySettings: !empty(securitySettings) ? securitySettings : null
+    publicNetworkAccess: enablePublicNetworkAccess ? 'Enabled' : 'Disabled'
```

Comment:
For this I'd suggest to follow what done for other modules: disable publicNetworkAccess by default if private endpoint is set. This would also involve to add an empty string as an allowed value for the publicNetworkAccess parameter. You can reference the implementation in the ML workspace module https://github.com/Azure/ResourceModules/blob/main/modules/Microsoft.DataFactory/factories/deploy.bicep#L199

Reply:
Updated in latest PR - PublicnetworkAccess now using `!empty(publicNetworkAccess) ? any(publicNetworkAccess) : (!empty(privateEndpoints) ? 'Disabled' : null)`